contracts/Tokens/Prime/PrimeLiquidityProvider.sol

```diff
@@ -0,0 +1,340 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity 0.8.13;
+
+import { SafeERC20Upgradeable, IERC20Upgradeable } from "@openzeppelin/contracts-upgradeable/token/ERC20/utils/SafeERC20Upgradeable.sol";
+import { AccessControlledV8 } from "@venusprotocol/governance-contracts/contracts/Governance/AccessControlledV8.sol";
+
+contract PrimeLiquidityProvider is AccessControlledV8 {
+    using SafeERC20Upgradeable for IERC20Upgradeable;
+
+    /// @notice The max token distribution speed
+    uint224 public constant MAX_DISTRIBUTION_SPEED = 1e18;
+
+    /// @notice exp scale
+    uint256 internal constant EXP_SCALE = 1e18;
+
+    /// @notice Address of the Prime contract
+    address public immutable prime;
+
+    /// @notice The rate at which token is distributed (per block)
+    mapping(address => uint256) public tokenDistributionSpeeds;
+
+    /// @notice The rate at which token is distributed to the Prime contract
+    mapping(address => uint256) public lastAccruedBlock;
+
+    /// @notice The token accrued but not yet transferred to prime contract
+    mapping(address => uint256) public tokenAmountAccrued;
+
+    /// @notice Initial balance at the time of token initialzation
+    mapping(address => uint256) public initialBalances;
+
+    /// @notice Is funds transfer paused to prime token
+    bool public fundsTransferpaused;
```

Comment:
`fundsTransferPaused`. I would change also the event associated

Reply:
changed `fundsTransferPaused` to `isFundsTransferPaused`
 0ce69629170233d16184a62eaec618ea8ef3c1a5

Reply:
I overlooked the `P` there, my mistake.